src/services/NodeMonitor.ts

```diff
@@ -307,7 +307,7 @@ export class NodeMonitor {
 			) {
 				return;
 			}
-			const nodeInx = this.nodeList.findIndex((addedNode) => addedNode.publicKey === node.publicKey);
+			const nodeInx = this.nodeList.findIndex((addedNode) => addedNode.host === node.host);
```

Comment:
nodeList should be a map, but we can leave it as is since we are moving off this code.